diff --git a/external/openglcts/modules/common/glcShaderNegativeTests.cpp b/external/openglcts/modules/common/glcShaderNegativeTests.cpp
--- a/external/openglcts/modules/common/glcShaderNegativeTests.cpp
+++ b/external/openglcts/modules/common/glcShaderNegativeTests.cpp
@@ -357,7 +357,8 @@ public:
         }
 
         // Compute shader, not available for GLES2 and GLES3
-        if (!glslVersionIsES(m_glslVersion) || m_glslVersion >= GLSL_VERSION_310_ES)
+        if ((glslVersionIsES(m_glslVersion) && m_glslVersion >= GLSL_VERSION_310_ES) ||
+            (!glslVersionIsES(m_glslVersion) && m_glslVersion >= GLSL_VERSION_430))
         {
             // Layout qualifier test
             args["QUALIFIER"] = layout_str;
diff --git a/external/openglcts/modules/gl/gl4cGPUShaderFP64Tests.cpp b/external/openglcts/modules/gl/gl4cGPUShaderFP64Tests.cpp
--- a/external/openglcts/modules/gl/gl4cGPUShaderFP64Tests.cpp
+++ b/external/openglcts/modules/gl/gl4cGPUShaderFP64Tests.cpp
@@ -3336,8 +3336,12 @@ void GPUShaderFP64Test2::deinit()
     if (0 != m_texture_id)
     {
         gl.bindTexture(GL_TEXTURE_2D, 0);
-        gl.bindImageTexture(0 /* unit */, 0 /* texture */, 0 /* level */, GL_FALSE /* layered */, 0 /* layer */,
-                            GL_READ_ONLY, GL_RGBA8);
+        if (glu::contextSupports(m_context.getRenderContext().getType(), glu::ApiType::core(4, 2)) ||
+            m_context.getContextInfo().isExtensionSupported("GL_ARB_shader_image_load_store"))
+        {
+            gl.bindImageTexture(0 /* unit */, 0 /* texture */, 0 /* level */, GL_FALSE /* layered */, 0 /* layer */,
+                                GL_READ_ONLY, GL_RGBA8);
+        }
         gl.deleteTextures(1, &m_texture_id);
         m_texture_id = 0;
     }
@@ -7671,7 +7675,7 @@ std::string GPUShaderFP64Test5::getVertexShaderBody(const _test_case &test_case)
     const std::string src_type_string   = Utils::getVariableTypeString(test_case.src_type);
 
     /* Add version preamble */
-    result << "#version 420\n"
+    result << "#version 410\n"
               "\n";
 
     /* Declare output variables. Note that boolean output variables are not supported, so we need
@@ -10837,7 +10841,7 @@ std::string GPUShaderFP64Test8::getFragmentShaderBody(const _test_case &test_case)
     std::stringstream result_sstream;
 
     /* Form the body */
-    result_sstream << "#version 420\n"
+    result_sstream << "#version 410\n"
                       "\n"
                       "void main()\n"
                       "{\n"
@@ -10909,7 +10913,7 @@ std::string GPUShaderFP64Test8::getGeometryShaderBody(const _test_case &test_case)
     std::stringstream result_sstream;
 
     /* Form the body */
-    result_sstream << "#version 420\n"
+    result_sstream << "#version 410\n"
                       "\n"
                       "layout(points)                 in;\n"
                       "layout(max_vertices=1, points) out;\n"
@@ -10936,7 +10940,7 @@ std::string GPUShaderFP64Test8::getTessellationControlShaderBody(const _test_case &test_case)
     std::stringstream result_sstream;
 
     /* Form the body */
-    result_sstream << "#version 420\n"
+    result_sstream << "#version 410\n"
                       "\n"
                       "layout(vertices=4) out;\n"
                       "\n"
@@ -10962,7 +10966,7 @@ std::string GPUShaderFP64Test8::getTessellationEvaluationShaderBody(const _test_case &test_case)
     std::stringstream result_sstream;
 
     /* Form the body */
-    result_sstream << "#version 420\n"
+    result_sstream << "#version 410\n"
                       "\n"
                       "layout(isolines) in;\n"
                       "\n"
@@ -10988,7 +10992,7 @@ std::string GPUShaderFP64Test8::getVertexShaderBody(const _test_case &test_case)
     std::stringstream result_sstream;
 
     /* Form the body */
-    result_sstream << "#version 420\n"
+    result_sstream << "#version 410\n"
                       "\n"
                       "void main()\n"
                       "{\n"
